test/constants.ts

```diff
@@ -0,0 +1,184 @@
+import Web3, { ContractInstance } from 'web3';
+import FakeProvider from 'web3-fake-provider';
+import BigNumber from 'bignumber.js';
+
+import {
+  DeferredTransactionWrapper,
+  ITxParams,
+  TypeChainContract
+} from '../types/typechain-runtime';
+
+import { AbiDefinition } from '@0xproject/types';
+
+/**
+ * Provides a harness to test autogenerated contract types
+ */
+export class TestContract<T extends TypeChainContract> {
+  private mockWeb3: Web3;
+  private mockCustomWeb3: Web3;
+
+  private mockWeb3Contract: ContractInstance;
+  private mockCustomWeb3Contract: ContractInstance;
+
+  constructor(contractName: string, private address: string, contractCode?: string) {
+    this.mockWeb3 = new Web3(new FakeProvider());
+
+    this.mockWeb3Contract = {
+      address: address,
+      abi: require(`../node_modules/market-solidity/build/contracts/${contractName}.json`)
+    };
+
+    spyOn(this.mockWeb3.eth, 'getCode').and.callFake(
+      (addr: string, cb: (err: Error, code: string) => void) => {
+        expect(addr).toEqual(address);
+        cb(null, contractCode || '0x234');
```

Comment:
@ConnorChristie -- So by lint errors I meant such as this one --
`null is not assignable to parameter of type Error`

Reply:
I am not sure I see that anywhere... This is all that shows in Travis and my local as well:

```
> tslint -t codeFrame 'types/**/*.ts' 'test/**/*.ts'
Warning: The 'await-promise' rule requires type information.
Warning: The 'no-unused-variable' rule requires type information.
Warning: The 'no-use-before-declare' rule requires type information.
Warning: The 'return-undefined' rule requires type information.
Warning: The 'no-floating-promises' rule requires type information.
Warning: The 'no-unnecessary-qualifier' rule requires type information.
Warning: The 'no-unnecessary-type-assertion' rule requires type information.
Warning: The 'strict-type-predicates' rule requires type information.
```

Reply:
@ConnorChristie -- Ahh. That was in the IDE actually. I use IntelliJ Idea. It isn't major, but maybe you can try configuring your IDE to use the `tslint.json` and `tsconfig.json` and confirm you're seeing the similar warnings on your end as well.